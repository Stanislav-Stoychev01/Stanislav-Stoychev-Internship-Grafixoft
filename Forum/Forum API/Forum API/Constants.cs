namespace Forum_API
{
    public static class Constants
    {
        public const string defaultRoleDescription = "default user";
        public const string dateFormat = "yyyy - MM - dd HH:mm:ss";
    }
}
