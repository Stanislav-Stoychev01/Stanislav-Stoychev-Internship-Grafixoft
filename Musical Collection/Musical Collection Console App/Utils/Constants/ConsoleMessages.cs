using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Musical_Collection_Console_App.Constants
{
    public class ConsoleMessages
    {
        public const string SuccessfulRegister = "You have registered succesfully";
        public const string SuccessfulLogin = "You have logged in succesfully";
        public const string UnsuccessfulLogin = "Invalid credentials";
        public const string InvalidAction = "The listener must be logged in first";
    }
}
