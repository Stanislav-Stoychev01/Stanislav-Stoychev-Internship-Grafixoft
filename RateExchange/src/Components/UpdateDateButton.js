import "../CSS/ButtonUpdateDate.css";
import { getCurrentDay } from "../Utils/Date";

function UpdateDateButton(props) {

    const currentDate = getCurrentDay();

    const onDateChange = () => {
        props.onDateChange !== undefined
        && props.onDateChange(currentDate);
    }

    return (
        props.date !== currentDate 
        && <div className="button-update-date-container">    
            <button className="button-update-date" onClick={onDateChange}>
                { props.buttonText }
            </button>
        </div>
    )
}

export default UpdateDateButton