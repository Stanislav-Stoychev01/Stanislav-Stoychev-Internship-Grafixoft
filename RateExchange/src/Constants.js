export const currencyList = ['USD', 'EUR', 'AUD', 'CAD', 'CHF', 'NZD', 'BGN'];

export const defaultCurrency = 'USD';


export const columnCount = 3;
export const firstColumn = 'First Column';
export const secondColumn = 'Second Column';
export const thirdColumn = 'Third Column';

export const firstColumnIndex = 0;
export const secondColumnIndex = 1;
export const thirdColumnIndex = 2;

export const key = 0;
export const value = 1;

export const dateString = 'date';
export const latest = 'latest';

export const inputLabel = "Input Currency Name:";
export const exchangeRateLabel = "Exchange Rates from";
export const longestSequenceMessage = "The longest sequence has length of ";

export const tableColumnNames = ['Conversion Rates below 1', 
'Conversion Rates between 1 and 1,5', 'Conversion Rates above 1,5'];

export const LoadingData = 'The data is currenly loading';