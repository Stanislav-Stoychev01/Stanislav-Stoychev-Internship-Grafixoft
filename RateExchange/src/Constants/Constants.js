export const CurrencyList = ['USD', 'EUR', 'AUD', 'CAD', 'CHF', 'NZD', 'BGN']
export const DefaultCurrency = 'USD'
