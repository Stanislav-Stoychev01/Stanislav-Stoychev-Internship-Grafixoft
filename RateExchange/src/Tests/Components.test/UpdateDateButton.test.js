import UpdateDateButton from "../../Components/UpdateDateButton";
import * as Constants from "../../Constants";
import Enzyme from 'enzyme';
import toJson from "enzyme-to-json";
import { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-17-updated';

Enzyme.configure({ adapter: new Adapter() });

it("renders dafault update date button", () => {

    const date = "2022-17-10";
    const { buttonText } = Constants;

    const props = {
        date: date, 
        buttonText: buttonText,
    };
    
    const wrapper = mount(<UpdateDateButton {...props} />);

    const wrapperToJson = toJson(wrapper);

    expect(wrapperToJson).toMatchSnapshot();
});