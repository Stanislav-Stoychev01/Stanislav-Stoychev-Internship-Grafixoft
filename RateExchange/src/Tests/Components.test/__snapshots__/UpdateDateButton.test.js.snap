// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders dafault update date button 1`] = `
<UpdateDateButton
  date="2022-17-10"
>
  <div
    className="button-update-date-container"
  >
    <button
      className="button-update-date"
      onClick={[Function]}
    />
  </div>
</UpdateDateButton>
`;
